### PR TITLE
rpm,deb,cmake: remove the lua submodule

### PR DESCRIPTION
**Pull request by yuvalif**
instead use the official lua-dev/el packages

Signed-off-by: Yuval Lifshitz <ylifshit@redhat.com>

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: ceph.spec.in [hunk 1]
```diff
@@ -179,6 +179,7 @@ BuildRequires:	xfsprogs
 BuildRequires:	xfsprogs-devel
 BuildRequires:	xmlstarlet
 BuildRequires:	nasm
+BuildRequires:  lua-devel
```
Comment (batrick) [inline ceph.spec.in:182]: How do we ensure this is Lua 5.3?

(Makes me wonder why not use Lua 5.4... but I suppose that's not available on debian/centos yet?)

Comment (tchaikov) [reply ceph.spec.in:182]: CentOS/RHEL8 only has lua 5.3 packaged. see https://centos.pkgs.org/8/centos-powertools-x86_64/lua-devel-5.3.4-11.el8.x86_64.rpm.html

> How do we ensure this is Lua 5.3?

could put
```
BuildRequires:  lua-devel >= 5.3
```

but i think `BuildRequires:  lua-devel` is good enough.

File: debian/control [hunk 2]
```diff
@@ -97,6 +97,7 @@ Build-Depends: cmake (>= 3.10.2),
 # Make-Check   xfsprogs,
 # Make-Check   xmlstarlet,
                nasm [amd64],
+               liblua5.3-dev,
```
Comment (tchaikov) [inline debian/control:100]: could you order this alphabetically?

**Review by tchaikov — APPROVED** [review]

**Review by tchaikov — CHANGES_REQUESTED** [review]
could you change the prefix of the title to something like

> rpm,deb,cmake: remove the lua submodule

?

**Comment by yuvalif** [comment]
> could you change the prefix of the title to something like
> 
> > rpm,deb,cmake: remove the lua submodule
> 
> ?

done

**Review by tchaikov — APPROVED** [review]

**Comment by yuvalif** [comment]
@tchaikov could you please approve the change to the submodule?

**Comment by tchaikov** [comment]
> @tchaikov could you please approve the change to the submodule?

if you could push it to ceph-ci to verify that it still builds. i can help merge it.

**Comment by yuvalif** [comment]
> > @tchaikov could you please approve the change to the submodule?
> 
> if you could push it to ceph-ci to verify that it still builds. i can help merge it.

doing that right now... will also run against teuthology to make sure mds is not broken
(lua in rgw and cls are covered by unit tests).

**Comment by yuvalif** [comment]
> > > @tchaikov could you please approve the change to the submodule?
> > 
> > 
> > if you could push it to ceph-ci to verify that it still builds. i can help merge it.
> 
> doing that right now... will also run against teuthology to make sure mds is not broken
> (lua in rgw and cls are covered by unit tests).

it builds ok in shaman: https://shaman.ceph.com/builds/ceph/wip-yuval-remove-lua-submodule/80223d8b3cc81f37b74f25b412252eb28d8f7e15/

@batrick do you know which teuthology suite I should run to verify that mds/mantle is not broken?

**Comment by batrick** [comment]
> > > > @tchaikov could you please approve the change to the submodule?
> > > 
> > > 
> > > if you could push it to ceph-ci to verify that it still builds. i can help merge it.
> > 
> > 
> > doing that right now... will also run against teuthology to make sure mds is not broken
> > (lua in rgw and cls are covered by unit tests).
> 
> it builds ok in shaman: https://shaman.ceph.com/builds/ceph/wip-yuval-remove-lua-submodule/80223d8b3cc81f37b74f25b412252eb28d8f7e15/
> 
> @batrick do you know which teuthology suite I should run to verify that mds/mantle is not broken?

Unfortunately we have no tests for mantle so I'm afraid "does it build" will have to suffice.

**Comment by yuvalif** [comment]
> > > > > @tchaikov could you please approve the change to the submodule?
> > > > 
> > > > 
> > > > if you could push it to ceph-ci to verify that it still builds. i can help merge it.
> > > 
> > > 
> > > doing that right now... will also run against teuthology to make sure mds is not broken
> > > (lua in rgw and cls are covered by unit tests).
> > 
> > 
> > it builds ok in shaman: https://shaman.ceph.com/builds/ceph/wip-yuval-remove-lua-submodule/80223d8b3cc81f37b74f25b412252eb28d8f7e15/
> > @batrick do you know which teuthology suite I should run to verify that mds/mantle is not broken?
> 
> Unfortunately we have no tests for mantle so I'm afraid "does it build" will have to suffice.

ok. so build is passing (the centos/crimson failure is unrelated).

**Pull request merged**